ext/gd/gd.c

```diff
@@ -3687,13 +3687,25 @@ PHP_FUNCTION(imageaffine)
 		if ((zval_affine_elem = zend_hash_index_find(Z_ARRVAL_P(z_affine), i)) != NULL) {
 			switch (Z_TYPE_P(zval_affine_elem)) {
 				case IS_LONG:
-					affine[i]  = Z_LVAL_P(zval_affine_elem);
+					affine[i] = Z_LVAL_P(zval_affine_elem);
+					if (affine[i] < INT_MIN || affine[i] > INT_MAX) {
```

Comment:
This guard could use `ZEND_LONG_EXCEEDS_INT()`, but for consistency it maybe better this way (and the compiler will optimize away anyway).